src/Umbraco.Cms.Integrations.Search.Algolia/Controllers/SearchController.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Algolia.Search.Models.Search;
+
+using Microsoft.AspNetCore.Mvc;
+
+using System.Text.Json;
+using Umbraco.Cms.Integrations.Search.Algolia.Builders;
+using Umbraco.Cms.Integrations.Search.Algolia.Migrations;
+using Umbraco.Cms.Integrations.Search.Algolia.Models;
+using Umbraco.Cms.Integrations.Search.Algolia.Services;
+using Umbraco.Cms.Web.BackOffice.Controllers;
+using Umbraco.Cms.Web.Common;
+using Umbraco.Cms.Web.Common.Attributes;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Controllers
+{
+    [PluginController("UmbracoCmsIntegrationsSearchAlgolia")]
+    public class SearchController : UmbracoAuthorizedApiController
+    {
+        private readonly IAlgoliaIndexService _indexService;
+
+        private readonly IAlgoliaSearchService<SearchResponse<Record>> _searchService;
+
+        private readonly IScopeService<AlgoliaIndex> _scopeService;
+
+        private readonly UmbracoHelper _umbracoHelper;
+
+        public SearchController(IAlgoliaIndexService indexService, IAlgoliaSearchService<SearchResponse<Record>> searchService, 
+            IScopeService<AlgoliaIndex> scopeService,
+            UmbracoHelper umbracoHelper)
+        {
+            _indexService = indexService;
+            
+            _searchService = searchService;
+
+            _scopeService = scopeService;
+
+            _umbracoHelper = umbracoHelper;
+        }
+
+        [HttpGet]
+        public IActionResult GetIndices()
+        {
+            var results = _scopeService.Get().Select(p => new IndexConfiguration
+            {
+                Id = p.Id,
+                Name = p.Name,
+                ContentData = JsonSerializer.Deserialize<List<ContentData>>(p.SerializedData)
+            });
+
+            return new JsonResult(results);
+        } 
+
+        [HttpPost]
+        public string SaveIndex([FromBody] IndexConfiguration index)
+        {
+            try
+            {
+                _scopeService.AddOrUpdate(new AlgoliaIndex
+                {
+                    Id = index.Id,
+                    Name = index.Name,
+                    SerializedData = JsonSerializer.Serialize(index.ContentData),
+                    Date = DateTime.Now
+                });
+
+                var payload = new List<Record>();
```

Comment:
I'm still not sure why we need this part.  It could be very expensive to find all content by document type so I think this better removed and this operation is solely about saving index definitions to the database.

It would be the publish content events that should be used to actually populate the indexes.

That said, there might be value in having a separate option for fully re-indexing a particular index (i.e. deleting all the values in there, and re-creating using the code you have here).  But that should be a separate button, like on the examine dashboard, with copy indicating that it could take a long time.